File: tests/fixtures/lang_samples/java/Nested.java
class Nested {
    int deeplyNested(int x) {
        if (x > 0) {
            for (int i = 0; i < x; i++) {
                if (i % 2 == 0) {
                    while (i > 0) {
                        try {
                            if (i == 5) {
                                return i;
                            }
                        } catch (Exception e) {
                        }
                        i--;
                    }
                }
            }
        }
        return 0;
    }

    int shallow(int x) {
        return x + 1;
    }
}

File: tests/fixtures/lang_samples/pascal/nested.pas
unit Nested;

interface

implementation

function DeeplyNested(X: Integer): Integer;
var
  I: Integer;
begin
  if X > 0 then
  begin
    for I := 0 to X do
    begin
      if I mod 2 = 0 then
      begin
        while I > 0 do
        begin
          if I = 5 then
          begin
            Result := I;
            Exit;
          end;
          Dec(I);
        end;
      end;
    end;
  end;
  Result := 0;
end;

function Shallow(X: Integer): Integer;
begin
  Result := X + 1;
end;

end.
